test/testGraphical.m

```diff
@@ -0,0 +1,44 @@
+function [ status ] = testGraphical( varargin )
+%TESTGRAPHICAL Runs the M2T test suite to produce graphical output
+%
+% This is quite a thin wrapper around testMatlab2tikz to run the test suite to
+% produce a PDF side-by-side report.
+%
+% Its allowed arguments are the same as those of testMatlab2tikz.
+%
+% Usage:
+%
+%     status = testGraphical(...) % gives programmatical access to the data
+%
+%     testGraphical(...); % automatically invokes makeLatexReport afterwards
+%
+% See also: testMatlab2tikz, testHeadless, makeLatexReport
+
+    status = testMatlab2tikz('actionsToExecute', @actionsToExecute, ...
+                             varargin{:});
+
+    if nargout == 0
+        makeLatexReport(status);
+    end
+
+end
+
+function status = actionsToExecute(status, ipp)
+    status = execute_plot_stage(status, ipp);
+    
+    if status.skip
+        return
```

Comment:
From looking at the code I'd say this is the cause for `status` missing some fields and making `texfile_addtest()` fail, when a test was skipped. Consequently, here `status.plotStage.error = 1;` and `status.tikzStage.error = 1;` should be set. For the current design `status.saveStage.texReference = '';` and `status.tikzStage.pdfFile = '';` is needed in addition.
Alternatively, `texfile_addtest` could check for `status.skip` and if true, skip adding the (non existent) content).